pkg/integrations/cadvisor/cadvisor_stub.go

```diff
@@ -0,0 +1,10 @@
+//go:build !linux
+// +build !linux
+
+package cadvisor //nolint:golint
+
+import "github.com/grafana/agent/pkg/integrations"
+
+func init() {
+	integrations.RegisterIntegration(integrations.NewStubConfig(name, "the cadvisor integration only works on linux; enabling it on other platforms will do nothing"))
+}
```

Comment:
One liner stub @rfratto 😎

Reply:
Love it :) 

You're losing out on the field checking here though (as you've noticed in the other comment. Another way of doing it is the following:

1. Define the Config struct for all platforms, no build tags. 
2. Have two implementations of New: one for Linux, one for everything else. 
3. The "everything else" implementation would do something like the following:

```go
func New(logger log.Logger, c *Config) (integrations.Integration, error) {
   level.Warn(logger).Log("msg", "cadvisor ...") 

   // return something that just implements a fake Run method, running 
   // until the ctx is canceled. 
   // 
   // This might already exist? 
   return integrations.NoOp, nil
}
```

(sorry, this is probably more straightforward but it didn't occur to me until now)

Reply:
Yeah, the problem with this one in particular, is that my config actually uses modules from cAdvisor, and it won't compile in anything but Linux.

I may have to circle back here on this integration, and try to prune out those bits from my config to clean this up. 🤔

Reply:
Ah, is that because you import `github.com/google/cadvisor/container`? Thankfully it looks like in this case you can replace `container.MetricSet` with a plain `map[string]interface{}` and validate later, but it's not very pretty. 

You might be able to change the stub to workaround the strict unmarshaling, but it wouldn't be pretty either, something like:

```go
// Unmarshal into a map so no field is rejected. 
var raw map[string]interface{}
if err := unmarshal(&raw); err != nil {
  return err 
}

// Convert the map back into YAML so we could unmarshal
// again (lol)
rawBytes, err := yaml.Marshal(raw) 
if err != nil {
  return err 
}

// Then do a final non-strict unmarshal into our stub struct.
return yaml.Unmarshal(rawBytes, (*plain)(c))
```

Reply:
I decided to remove the `StubConfig` since it was a bad code smell. The *right* pattern here is to design a config that can be used on any platform.

For cAdvisor, I added an accessor func on the config struct that "does the magic" to convert the string slice to the desired native cAdvisor type, and it's only built on windows.

I think this is a pretty workable solution and pattern for integrations like this.